lib/membrane_rtc_engine/endpoints/webrtc/encoding_selector.ex

```diff
@@ -0,0 +1,193 @@
+  chosen whenever it is active. Can be changed with `target_encoding/2`.
+  """
+  @spec new([Track.variant()], Track.variant() | nil) :: t()
+  def new(encodings, initial_target_encoding \\ nil) do
+    %__MODULE__{all_encodings: encodings, target_encoding: initial_target_encoding}
+  end
+
+  @doc """
+  Marks given `encoding` as inactive.
+
+  Returns new selector and encoding to request
+  or `nil` if there are no changes needed.
+  """
+  @spec encoding_inactive(t(), Track.variant()) :: {t(), Track.variant() | nil}
+  def encoding_inactive(selector, encoding) do
+    selector = %__MODULE__{
+      selector
+      | active_encodings: List.delete(selector.active_encodings, encoding)
+    }
+
+    next_encoding = get_next_encoding(selector.active_encodings)
+
+    cond do
+      selector.current_encoding == encoding ->
+        selector = %__MODULE__{selector | current_encoding: nil}
+        select_encoding(selector, next_encoding)
+
+      selector.queued_encoding == encoding and selector.current_encoding == nil ->
+        selector = %__MODULE__{selector | queued_encoding: nil}
+        select_encoding(selector, next_encoding)
+
+      selector.queued_encoding == encoding ->
+        selector = %__MODULE__{selector | queued_encoding: nil}
+        {selector, nil}
+
+      true ->
+        {selector, nil}
+    end
+  end
+
+  @doc """
+  Marks given `encoding` as active.
+
+  See `encoding_inactive/2` for the meaning
+  of return values.
+  """
+  @spec encoding_active(t(), Track.variant()) :: {t(), Track.variant() | nil}
+  def encoding_active(selector, encoding) do
+    selector = %__MODULE__{
+      selector
+      | active_encodings: selector.active_encodings ++ [encoding]
+    }
+
+    next_encoding = get_next_encoding(selector.active_encodings)
+
+    cond do
+      selector.current_encoding == selector.target_encoding ->
+        {selector, nil}
+
+      selector.queued_encoding == selector.target_encoding ->
+        {selector, nil}
+
+      selector.target_encoding == encoding ->
+        select_encoding(selector, encoding)
+
+      true ->
+        select_encoding(selector, next_encoding)
+    end
+  end
+
+  @doc """
+  Sets currently used encoding.
+
+  Should be called when encoding change happens
+  i.e. after receiving `TrackVariantSwitched` event.
+  """
+  @spec current_encoding(t(), Track.variant()) :: t()
+  def current_encoding(%__MODULE__{queued_encoding: encoding} = selector, encoding) do
+    %__MODULE__{selector | current_encoding: encoding, queued_encoding: nil}
+  end
+
+  def current_encoding(%__MODULE__{} = selector, encoding) do
+    %__MODULE__{selector | current_encoding: encoding}
+  end
+
+  @doc """
+  Prioritizes encoding.
+
+  Prioritized encoding will be chosen whenever it
+  is active. See `encoding_inactive/2` for the
+  meaning of return values.
+  """
+  @spec target_encoding(t(), Track.variant()) :: {t(), Track.variant() | nil}
+  def target_encoding(selector, encoding) do
```

Comment:
`target_encoding` sets given encoding as our goal. This function is called on client request. E.g. client can point out that  `:medium` is its target encoding. If it is inactive, we will remember user choice (as target encoding) and when it becomes active we will request it. In other words this is encoding that we will chose whenver it is active.

Reply:
The trouble is that this is described inconsistently - I think we should consistently use either `target` or `priority`.
The way you and the docs describe it, the function should be named `prioritize_encoding`.
If sticking to target, then I feel like a verb is missing from the name. It might be just me, but I'd expect it to be named `set_target_encoding`